funsor/ops/op.py

```diff
@@ -8,25 +8,66 @@
 from multipledispatch import Dispatcher
 
 
+class WeakPartial:
```

Comment:
nit: consider moving `WeakPartial` to `funsor.util`?

Reply:
done